shell/src/Luna/Shell.hs

```diff
@@ -139,33 +147,129 @@ stdlibPath = do
                 <> " environment variable"
     return stdPath
 
+forceImports :: Imports -> [Maybe [CompileError]]
+forceImports imp =
+    let !functions = Map.map (view Function.documentedItem)
+                   $ view importedFunctions imp
+        !methods   = map (view Function.documentedItem)
+                   $ concatMap (Map.elems . view Class.methods)
+                   $ Map.elems
+                   $ Map.map (view Function.documentedItem)
+                   $ view importedClasses imp
+    in flip map (Map.elems functions ++ methods) $ \case
+        Left errors -> Just errors
+        Right     _ -> Nothing
+
+forceCompilation :: Project.CompiledModules -> IO ()
+forceCompilation (Project.CompiledModules m p) = do
+    let !errors = P.concat $ map forceImports (p : Map.elems m)
+    case catMaybes errors of
+        []  -> return ()
+        err -> throwM $ CompilationError $ P.concat err
+
+data CompilationError = CompilationError [CompileError]
+    deriving Show
+
+instance Exception CompilationError where
+    displayException (CompilationError errors) =
+        convert $ Layout.concatLineBlock $ Layout.render $ formatErrors errors
+
+data Options = Options { _projectDirectory      :: Maybe FilePath
+                       , _exhaustiveCompilation :: Bool
+                       , _fileWatch             :: Bool
+                       }
+
+makeLenses ''Options
+
+options :: OptParse.Parser Options
+options = Options
+      <$> OptParse.optional (OptParse.argument OptParse.str (
+              OptParse.metavar "PROJECT_DIR"
+           <> OptParse.help "Path to a project"))
+      <*> OptParse.switch (
+              OptParse.long "exhaustive"
+           <> OptParse.help ("Enables exhaustive compilation "
+                          <> "of all sources and functions"))
+      <*> OptParse.switch (
+              OptParse.long "file-watch"
+           <> OptParse.help "Enables recompilation on a file change")
+
+
 main :: IO ()
-main = do
-    mainPath'    <- getCurrentDirectory
-    mainPath     <- Path.parseAbsDir mainPath'
-    let mainName  = Project.getProjectName mainPath
-    stdPath      <- stdlibPath
-    (_, std)     <- Project.prepareStdlib  (Map.fromList [("Std", stdPath)])
+main = shell =<< OptParse.execParser opts
+    where
```

Comment:
could we move this `where` to previous like? In all the code we keep one of two styles:

```
foo bar baz where
   ...
```

or

```
foo bar baz
    where ...
          ...
```